packages/axelar-wasm-std/src/msg_id/starknet_field_element_event_index.rs

```diff
@@ -0,0 +1,284 @@
+use core::fmt;
+use std::fmt::Display;
+use std::str::FromStr;
+
+use cosmwasm_std::HexBinary;
+use error_stack::Report;
+use lazy_static::lazy_static;
+use regex::Regex;
+use serde_with::DeserializeFromStr;
+use starknet_types_core::felt::Felt;
+
+use super::Error;
+use crate::nonempty;
+use crate::utils::does_felt_overflow_from_str;
+
+#[derive(Debug, DeserializeFromStr, Clone)]
+pub struct FieldElementAndEventIndex {
+    pub tx_hash: Felt,
```

Comment:
From what I've seen the `Felt` type doesn't guarantee validity. I would prefer if you'd implement something like a `ValidatedFelt` type, similar to nonempty::String, which makes sure that all checks have passed. Then you can also easily implement `TryFrom`s from byte slices and strings and don't have to do the multiple `does_felt_overflow_from_X` implementations and `FieldElementAndEventIndex` is always guaranteed to be valid